src/odemis/odemisd/modelgen.py

```diff
@@ -792,4 +853,50 @@ def get_instantiables(self, instantiated=None):
                 comps.add(n)
 
         return comps
+            
+    def _parse_settings(self, settings_file):
+        """
+        Parse settings file and return persistent properties and metadata in useful format.
+        settings_file (str): path to settings file
+        return (list of 2 nested dicts): persistent properties and persistent metadata for
+          each component in the settings file.
+          props[comp][prop_name] --> value, mds[comp][md_name] --> value
```

Comment:
actually, it should be open the entire time since the beginning, so does it even make sense to check for it? If there is an error because the file doesn't exist, it should fail before.

Reply:
Indeed, I hadn't realized that. Once the docstring is updated (to indicate the function receives an _opened_ file, it should be good enough.